lib/layer.js

```diff
@@ -41,7 +40,7 @@ function Layer (path, options, fn) {
   this.name = fn.name || '<anonymous>'
   this.params = undefined
   this.path = undefined
-  this.regexp = pathRegexp((opts.strict ? path : loosen(path)), this.keys, opts)
+  this.regexp = pathRegexp.match((opts.strict ? path : loosen(path)), opts)
```

Comment:
Can we get rid of the `_slash` behavior? If it's a real perf win I'll move it into `path-to-regexp`?

Reply:
Yeah that would be great. I am not sure it is a perf win anymore now that we have removed the regexp parts. For now, all the tests are passing locally (after the updates I just pushed) other than the array ones. Once you get that added, we can probably publish since the `_slash` optimization can be changed in a minor.

Reply:
So while I was working through the other test fixes after this I figured out what it was that fixed.

```
const fn = match('/', {
  end: false
});
console.log(fn('/foo'));
```

This case is what `.use` does, and it means that the `.use(mw)` call does not match the route. To me this behavior just seems broken and like we could replace it with much more simple ways, but for now I am trying to do the minimal change set to get the update done. I brought it back after trying twice to delete it lol, so was not minimal in the amount of work, but it reduces the changes in this PR a bit.

Are we alright with this compromise for now until we can get the release out and then do more fixes in minors?